### DISCUSSION TITLE
support kubevirt as a infrastructure provider

### DISCUSSION THREAD

**Discussion by HenryGuo1019** [rancher-sandbox/cluster-api-provider-rke2#318] [Q&A]
How can i build a air gapped vm image so that rke2 can bootstrap automatically?

**Comment by belgaied2** [comment]
@HenryGuo1019 you can take inspiration from [here](https://github.com/rancher-sandbox/cluster-api-provider-rke2/tree/main/samples/docker/air-gapped/image-building) for image building in Air Gapped mode.
The example shown there is based on Docker, but the same principles apply.

The idea is:
- Use a base OS that supports cloud-init or ignition.
- Put prerequired files for your RKE2 version in a specific target folder ( `/opt/rke2-artifacts` ) (TAR files that you can download from the RKE2 release page) 
- Specify `true` in `RKE2ConfigTemplate.spec.agentConfig.airGapped`.

More details in the page above.
